From 5dbbda543537ede4253a5c1cfc6a9d3d84058191 Mon Sep 17 00:00:00 2001
From: Hadrien Croubois <hadrien.croubois@gmail.com>
Date: Mon, 8 Mar 2021 19:42:34 +0100
Subject: [PATCH] Make ERC1155.uri public (PR 2576)

---
 token/ERC1155/ERC1155.sol | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/token/ERC1155/ERC1155.sol b/token/ERC1155/ERC1155.sol
index eccc320..40d18a5 100644
--- a/token/ERC1155/ERC1155.sol
+++ b/token/ERC1155/ERC1155.sol
@@ -72,7 +72,7 @@ contract ERC1155 is Context, ERC165, IERC1155, IERC1155MetadataURI {
      * Clients calling this function must replace the `\{id\}` substring with the
      * actual token type ID.
      */
-    function uri(uint256) external view virtual override returns (string memory) {
+    function uri(uint256) public view virtual override returns (string memory) {
         return _uri;
     }
 
-- 
2.17.1

